public/app/plugins/datasource/loki/APP_PLUGIN_DOCUMENTATION.md

```diff
@@ -0,0 +1,148 @@
+
+### Fetching Loki label names
+
+`datasource.languageProvider.fetchLabels()` can be used to fetch label names from connected Loki data source. Labels are essential for organizing and filtering log data. You can use this method to retrieve labels, providing a foundation for various data manipulation tasks within your app plugin.
+
+```ts
+/**
+ * Fetch all label keys
+ * This asynchronous function is designed to retrieve all available label keys from the data source.
+ * It returns a promise that resolves to an array of strings containing the label keys.
+ *
+ * @returns A promise containing an array of label keys.
+ * @throws An error if the fetch operation fails.
+ */
+async function fetchLabels(): Promise<string[]>;
+
+/**
+ * Example usage:
+ */
+
+try {
+  const labelKeys = await fetchLabels();
+  console.log(labelKeys);
+} catch (error) {
+  console.error(`Error fetching label keys: ${error.message}`);
+}
+```
+
+### Fetching Loki label values
+
+The `datasource.languageProvider.fetchLabelValues()` method is designed for fetching label values. This API enables you to retrieve the values associated with a particular label.
+
+```ts
+/**
+ * Fetch label values
+ *
+ * This asynchronous function fetches values associated with a specified label name.
+ * It returns a promise that resolves to an array of strings containing the label values.
+ *
+ * @param labelName - The name of the label for which you want to retrieve values.
+ * @returns A promise containing an array of label values.
+ * @throws An error if the fetch operation fails.
+ */
+async function fetchLabelValues(labelName: string): Promise<string[]>;
+
+/**
+ * Example usage:
+ */
+
+const labelName = 'job';
+try {
+  const values = await fetchLabelValues(labelName);
+  console.log(values);
+} catch (error) {
+  console.error(`Error fetching label values: ${error.message}`);
+}
+```
+
+### Fetching Loki labels for a specified selector
+
+`datasource.languageProvider.fetchSeriesLabels` can be used to fetch available labels for a given stream selector.
+
+```ts
+/**
+ * Fetch series labels for a selector
+ *
+ * This method fetches labels for a given stream selector, such as `{job="grafana"}`.
+ * It returns a promise that resolves to a record mapping label names to their corresponding values.
+ *
+ * @param streamSelector - The stream selector for which you want to retrieve labels.
+ * @returns A promise containing a record of label names and their values.
+ * @throws An error if the fetch operation fails.
+ */
+async function fetchSeriesLabels(streamSelector: string): Promise<Record<string, string[]>>;
+
+/**
+ * Example usage:
+ */
+const streamSelector = '{job="grafana"}';
+try {
+  const labels = await fetchSeriesLabels(streamSelector);
+  console.log(labels);
+} catch (error) {
+  console.error(`Error fetching labels: ${error.message}`);
+}
+```
+
+### Fetching Loki parser and label keys for log stream based on sampled lines
```

Comment:
```suggestion
### Detecting Loki parser and label keys for log stream based on sampled lines
```

I think it's important to use "Detecting" or "Inferring" because it's what the code does, unless I misremember.